### PR TITLE
WFCORE-3167 option to filter out aged out patches from the patching h…

### PR DESCRIPTION
**Pull request by aloubyansky**
…istory

### CODE REVIEW

**Comment by aloubyansky** [comment]
retest this please

File: patching/src/main/resources/help/patch.txt [hunk 1]
```diff
@@ -96,11 +96,16 @@ ACTION: history
...
     patch history
       [--patch-stream=<patch-stream-name>]
+      [--exclude-agedout]
       [--host=<host>]
       [--distribution=<target-distribution>]
       [--module-path=<modules-dir>]
       [--bundle-path=<bundles-dir>]
 
+      --exclude-agedout  - Optional argument, if present patches that have been
+                           aged out with ageout-history operation will not be
```
Comment (bstansberry) [inline patching/src/main/resources/help/patch.txt]: s/with/with the/g

File: patching/src/main/resources/org/jboss/as/patching/management/LocalDescriptions.properties [hunk 2]
```diff
@@ -62,6 +62,7 @@ patching.rollback-last.override=List of the miscellaneous items that can be over
 patching.rollback-last.preserve=List of the miscellaneous items that must be preserved and not modified by rolling back the patch.
 patching.rollback-last.reset-configuration=The rollback operation will update the installation configurations and reset the snapshots that were taken when the patch was applied.
 patching.show-history=Show history of the patches that have been applied.
+patching.show-history.exclude-agedout=Whether to filter out patches that have been removed by ageout-history operation.
```
Comment (bstansberry) [inline patching/src/main/resources/org/jboss/as/patching/management/LocalDescriptions.properties]: s/by/by the/g

**Comment by jfdenise** [comment]
@bstansberry , @aloubyansky , I have included this patch (with adaptation for the new patch command implementation) in this PR: https://github.com/jfdenise/wildfly-core/commits/4.0-aesh1.0-operators

You can see how to add a boolean value option with the new API: https://github.com/jfdenise/wildfly-core/commit/8a800ef1a7f6463aa28606222be5f1d0f23143dd#diff-3eeecd734c40c0208e9415d4da39888cR34
and for help: https://github.com/jfdenise/wildfly-core/commit/8a800ef1a7f6463aa28606222be5f1d0f23143dd#diff-d31591675d95f9be23f85f0285fd2bb8R100

I have also applied the original changes to the PatchHandler (for reference, at some point I will remove this class, the help file has been already removed):https://github.com/jfdenise/wildfly-core/commit/8a800ef1a7f6463aa28606222be5f1d0f23143dd#diff-563081ef982b874e3c49546e9e9584daR116

@bstansberry, I can still see an "agedout" usage (https://github.com/jfdenise/wildfly-core/commit/8a800ef1a7f6463aa28606222be5f1d0f23143dd#diff-a8c3fab5aac451c2d9c48f9b145f47b8R46) that I didn't fix, shouldn't we change it too to be consistent?

**Comment by bstansberry** [comment]
@aloubyansky @jfdenise I'm closing this as it's in the other patch.

**Pull request closed without merging**